src/psyclone/psyir/backend/fortran.py

```diff
@@ -231,6 +232,85 @@ def _reverse_map(op_map):
     return mapping
 
 
+def get_fortran_operator(operator):
+    '''Determine the Fortran operator that is equivalent to the provided
+    PSyIR operator. This is achieved by reversing the Fparser2Reader
+    maps that are used to convert from Fortran operator names to PSyIR
+    operator names.
+
+    :param operator: a PSyIR operator.
+    :type operator: \
+        :py:class:`psyclone.psyir.nodes.Operation.Operator`
+
+    :returns: the Fortran operator.
+    :rtype: str
+
+    :raises KeyError: if the supplied operator is not known.
+
+    '''
+    unary_mapping = _reverse_map(Fparser2Reader.unary_operators)
+    if operator in unary_mapping:
+        return unary_mapping[operator].upper()
+
+    binary_mapping = _reverse_map(Fparser2Reader.binary_operators)
+    if operator in binary_mapping:
+        return binary_mapping[operator].upper()
+
+    nary_mapping = _reverse_map(Fparser2Reader.nary_operators)
+    if operator in nary_mapping:
+        return nary_mapping[operator].upper()
+    raise KeyError()
+
+
+def is_fortran_intrinsic(fortran_operator):
+    '''Determine whether the supplied Fortran operator is an intrinsic
+    Fortran function or not.
+
+    :param str fortran_operator: the supplied Fortran operator.
+
+    :returns: true if the supplied Fortran operator is a Fortran \
+        intrinsic and false otherwise.
+
+    '''
+    if fortran_operator in FORTRAN_INTRINSICS:
```

Comment:
it can also just be `return fortran_operator in FORTRAN_INTRINSICS`

Reply:
Oh yes :-).